### PR TITLE
feat: Add 'who has access' to sharing modal

### PR DESCRIPTION
**Pull request by GoOz**
Add the view that displays the access list

@TODO get the access list from the API.

### CODE REVIEW

**Comment by GoOz** [comment]
[(2) [PHO] Onglet "Who has access" dans la modale de partage en lecture seule](https://trello.com/c/dpz2Hml1/366-2-pho-onglet-who-has-access-dans-la-modale-de-partage-en-lecture-seule)

**Comment by y-lohse** [comment]
So far so good! But I guess we'll amend this PR with the logic bits?

**Comment by GoOz** [comment]
This is not for merging purpose right now, just to be reviewed in advance 😛 
In case i didn't take the right path from the beginning.

**Comment by enguerran** [comment]
I added some functions, but faced an error on `GET /sharings/:id` with `:id` a sharing's id found in the permissions sets that is returned by `GET /permissions/doctype/io.cozy.files/sharedWithOthers`... I asked @Ljinod for a tips, and we will added the information in the [FAQ: Great! I know that my resource is shared. Can I have more information regarding the sharing?](https://github.com/Ljinod/cozy-stack/blob/PR-FAQ-sharings/docs/sharing.md#great-i-know-that-my-resource-is-shared-can-i-have-more-information-regarding-the-sharing).

**Comment by enguerran** [comment]
@GoOz @y-lohse, @goldoraf I think I'm done with this. Could you review my changes?

Note: all logic must go into `redux-cozy-api`, but for now, I write it right in the `sharing/index.js` lib file. As you can see if you run the code, there is a wrong behavior in the sharing modal as if you switch tab before fetching all needed data, it will go back to the first tab. That is not related to this feature, so I let it as this. But we should consider to refactor it later.

File: src/ducks/sharing/index.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+  findPermissionSets(ids, doctype, SHARED_BY_LINK, 'collection').then(sets => sets.map(set => set.attributes.permissions['collection'].values[0]))
 
 export const filterSharedWithMeDocuments = (ids, doctype) =>
-  findPermissionSets(ids, doctype, SHARED_WITH_ME).then(sets => sets.map(set => set.attributes.permissions.collection.values[0]))
+  findPermissionSets(ids, doctype, SHARED_WITH_ME, 'collection').then(sets => sets.map(set => set.attributes.permissions['collection'].values[0]))
 
-export const findPermSet = (id, doctype) =>
-  findPermissionSets([id], doctype, SHARED_BY_LINK).then(sets => sets.length === 0 ? undefined : sets[0])
+export const filterSharedWithOthersDocuments = (id, doctype) =>
+  findPermissionSets([id], doctype, SHARED_WITH_OTHERS, 'rule0')
```
Comment (y-lohse) [inline src/ducks/sharing/index.js]: Are you sure it's safe to use `rule0` here? IIRC there can be multiple rules in a permission set, and there's no guarantee that the ule we're looking for is the first one?

File: src/components/Recipient.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import styles from '../styles/recipient'
+
+import React from 'react'
+import { translate } from 'cozy-ui/react/I18n'
+import classNames from 'classnames'
+
+export const Recipient = ({ t, name, url, status }) => {
+  return (
+    <div className={classNames(styles['pho-recipient'])}>
```
Comment (goldoraf) [inline src/components/Recipient.jsx]: Using `classNames` is not necessary here, as it's only single classes. It'd be more readable without.

Comment (enguerran) [inline src/components/Recipient.jsx]: A vestige, I guess. I'll remove it.

File: src/ducks/sharing/index.js (hunk 1 above)
Comment (enguerran) [inline src/ducks/sharing/index.js]: Absolutely not safe, I agree.
But sharing is a cutting-edge feature of the `cozy-stack`, and as said we will need to refactor the code to organize it in real usable libs. So I offer to let this very ugly instruction and wait for a bug or a refactor to get rid of it.

**Pull request merged**